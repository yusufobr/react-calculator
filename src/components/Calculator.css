td {
  text-align: center;
  padding-bottom: 2px;
  padding-right: 2px;
  width: 25%;
}

td:hover {
  background-color: #00000008;
}

input[type="text"] {
  height: 50px;
}

.operator {
  background-color: rgb(255, 146, 82);
}
