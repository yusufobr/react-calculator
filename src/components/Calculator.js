import React from 'react';
import './Calculator.css';

export default function Calculator() {
  return (
    <div className="text-[#333] text-sm font-bold p-12">
      <table className="h-[300px] w-[350px] mx-auto bg-[#E0E0E0]">
        <thead>
          <tr>
            <td colSpan="4">
              <input type="text" value="0" className="w-full pr-1 text-right text-xl font-normal text-white bg-[#858693]" />
            </td>
          </tr>
        </thead>
        <tbody>
          <tr>
            <td>
              <input type="button" value="AC" />
            </td>
            <td>
              <input type="button" value="+/-" />
            </td>
            <td>
              <input type="button" value="%" />
            </td>
            <td className="operator">
              <input type="button" value="รท" />
            </td>
          </tr>
          <tr>
            <td>
              <input type="button" value="7" />
            </td>
            <td>
              <input type="button" value="8" />
            </td>
            <td>
              <input type="button" value="9" />
            </td>
            <td className="operator">
              <input type="button" value="x" />
            </td>
          </tr>
          <tr>
            <td>
              <input type="button" value="4" />
            </td>
            <td>
              <input type="button" value="5" />
            </td>
            <td>
              <input type="button" value="6" />
            </td>
            <td className="operator">
              <input type="button" value="-" />
            </td>
          </tr>
          <tr>
            <td>
              <input type="button" value="1" />
            </td>
            <td>
              <input type="button" value="2" />
            </td>
            <td>
              <input type="button" value="3" />
            </td>
            <td className="operator">
              <input type="button" value="+" />
            </td>
          </tr>
          <tr>
            <td colSpan="2">
              <input type="button" value="0" />
            </td>
            <td>
              <input type="button" value="." />
            </td>
            <td className="operator">
              <input type="button" value="=" />
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  );
}
