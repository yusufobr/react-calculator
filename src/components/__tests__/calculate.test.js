import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import calculate from '../../logic/calculate';
import Calculator from '../Calculator';

describe('calculate', () => {
  test('returns an empty object when buttonName is "AC"', () => {
    const result = calculate({}, 'AC');
    expect(result).toMatchSnapshot();
  });

  test('updates "next" with a number when no operation is set', () => {
    const obj = { next: '5', total: null, operation: null };
    const result = calculate(obj, '2');
    expect(result).toMatchSnapshot();
  });

  test('appends a number to "next" when an operation is set', () => {
    const obj = { next: '5', total: null, operation: '+' };
    const result = calculate(obj, '2');
    expect(result).toMatchSnapshot();
  });

  test('performs the operation when "next" and an operation are set', () => {
    const obj = { next: '5', total: '10', operation: 'x' };
    const result = calculate(obj, '=');
    expect(result).toMatchSnapshot();
  });
});

describe('Calculator Component', () => {
  test('simulates user interaction', () => {
    const { getByText } = render(<Calculator />);

    const button1 = getByText('1');
    fireEvent.click(button1);

    const buttonPlus = getByText('+');
    fireEvent.click(buttonPlus);

    const button5 = getByText('5');
    fireEvent.click(button5);

    const buttonEqual = getByText('=');
    fireEvent.click(buttonEqual);

    const result = getByText('6');

    expect(result).toBeInTheDocument();
  });
});
