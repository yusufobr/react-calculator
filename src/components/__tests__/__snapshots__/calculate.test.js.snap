// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`calculate appends a number to "next" when an operation is set 1`] = `
Object {
  "next": "52",
  "operation": "+",
  "total": null,
}
`;

exports[`calculate performs the operation when "next" and an operation are set 1`] = `
Object {
  "next": null,
  "operation": null,
  "total": "50",
}
`;

exports[`calculate returns an empty object when buttonName is "AC" 1`] = `
Object {
  "next": null,
  "operation": null,
  "total": null,
}
`;

exports[`calculate updates "next" with a number when no operation is set 1`] = `
Object {
  "next": "52",
  "total": null,
}
`;
